### PR TITLE
micropython support for LED animations

### PR DESCRIPTION
**Pull request by dwalton76**
For #633 

Tested python3 and micropython via
```
from ev3dev2.led import Led, Leds
from time import sleep
import logging
import sys

# logging
logging.basicConfig(level=logging.DEBUG,
                    format="%(asctime)s %(levelname)5s: %(message)s")
log = logging.getLogger(__name__)

my_leds = Leds()

for x in range(4):
    if x == 0:
        my_leds.animate_police_lights('RED', 'GREEN')
    elif x == 1:
        my_leds.animate_flash('AMBER')
    elif x == 2:
        my_leds.animate_cycle(('RED', 'GREEN', 'AMBER'))
    elif x == 3:
        my_leds.animate_rainbow()

my_leds.reset()
```

### CODE REVIEW

File: ev3dev2/stopwatch.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 try:
     import datetime as dt
     micropython = False
-except ImportError:
```
Comment (WasabiFan) [inline ev3dev2/stopwatch.py:8]: We sure we want to do that?

File: ev3dev2/stopwatch.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
...
+except Exception:
     import utime
     micropython = True
```
Comment (WasabiFan) [inline ev3dev2/stopwatch.py]: isn't there already a function we have to check this? And/or a uPy builtin?

File: ev3dev2/stopwatch.py (hunk 1 above)
Comment (dwalton76) [reply ev3dev2/stopwatch.py:8]: yeah it is a bit odd but on my laptop (ubuntu) it raises `ImportError` but on ev3dev it raised something else (`AttributeError` maybe) so to play it safe I just caught all Exceptions.

File: ev3dev2/stopwatch.py (hunk 2 above)
Comment (dwalton76) [reply ev3dev2/stopwatch.py]: ah you are right, we have
```
def is_micropython():
    return sys.implementation.name == "micropython"
```

Will use that

**Review by WasabiFan — APPROVED** [review]
Looks good to me, thanks for figuring these things out!

**Pull request merged**